app/models/wakes/location.rb

```diff
@@ -3,17 +3,24 @@ class Wakes::Location < ActiveRecord::Base
   include Wakes::Metrics::GoogleAnalyticsPageviews
   include Wakes::Metrics::Facebook
 
-  validates :path, :format => { :with => %r{\A\/} }, :uniqueness => true
+  validates :path, :format => { :with => %r{\A\/} }, :uniqueness => { :scope => :host }
   belongs_to :resource, :foreign_key => :wakes_resource_id, :inverse_of => :locations, :touch => true
 
   scope :canonical, -> { where(:canonical => true) }
   scope :legacy, -> { where(:canonical => false) }
 
   def label
-    path
+    (host + path if host.present?) ||
```

Comment:
@chrismaddocks what about something like `path_or_url.sub(%r{^https?://}, '')`? Basically, remove the protocol if it happens to be there.